### DISCUSSION TITLE
Hide plugin class filenames in new node selector

### DISCUSSION THREAD

**Discussion by innovate-invent** [godotengine/godot-proposals#9423] [Editor]
As a plugin developer I am having to use the [add_custom_type()](https://docs.godotengine.org/en/stable/classes/class_editorplugin.html#class-editorplugin-method-add-custom-type) function over the `class_name` and `@icon` features of GDScript if I don't want the sub-optimal user experience of the filename of the script to be included in the new node selector.

![with_filename](https://github.com/godotengine/godot-proposals/assets/1768820/22e37181-7341-4e47-a7bb-6459ca59c817)

Ideally the experience should be consistent regardless of using `add_custom_type` vs `class_name`. I propose that the filename be moved to a tool-tip of a new "plugin" icon right justified to the node name. Below is a mock of the proposed UI:

![mock](https://github.com/godotengine/godot-proposals/assets/1768820/7de55c9b-e41a-4ff5-841a-6070493d1186)

Additionally, it would be more useful to the user to display the plugin name within the tool-tip along with or instead of the filename.

This is a continuation of the chat conversation here: https://chat.godotengine.org/channel/editor?msg=hsdk5fzi6eKjCTqhQ

**Comment by Calinou** [comment]
I've opened a pull request to implement this: https://github.com/godotengine/godot/pull/90548
